src/components/ChatEntry.js

```diff
@@ -1,22 +1,38 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
+
+const ChatEntry = ({id, sender, body, timeStamp, liked, heartClick}) => {
+
+  const clickLikedHeart = () => {
+    heartClick(
+      {id:id,
+       sender:sender,
+       body:body, 
+       timeStamp:timeStamp, 
+       liked:!liked})};
 
-const ChatEntry = (props) => {
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{body}</p>
+        <p className="entry-time"><TimeStamp time= {timeStamp}></TimeStamp></p>
+        <button className="like" onClick={clickLikedHeart}>{liked ? '❤️': '🤍'}</button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number.isRequired,
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool,
+  heartClick: PropTypes.func
```

Comment:
missing `.isRequired` for these last 2 PropTypes

Reply:
oh wait! i remember you saying in the wrap-up session that tests were failing unless you removed .isRequired. Disregard this comment